packages/client-core/src/systems/ui/AvatarDetailView.tsx

```diff
@@ -33,10 +38,38 @@ function createAvatarDetailState(id: string) {
 }
 
 type CharacterDetailState = ReturnType<typeof createAvatarDetailState>
-
+interface TypingDetail {
+  typing: Boolean
+  userId: String
+}
 const CharacterDetailView = () => {
+  const { t } = useTranslation()
+  addActionReceptor(Engine.currentWorld.store, userTypingActionReceptor)
   const detailState = useXRUIState() as CharacterDetailState
   const userState = useUserState()
+  const [userTypingDetail, setUserTypingDetail] = React.useState<TypingDetail>()
   const user = userState.layerUsers.find((user) => user.id.value === detailState.id.value)
-  return user ? <div style={styles.avatarName as {}}>{user.name.value}</div> : <div></div>
+
+  function userTypingActionReceptor(action) {
+    const world = Engine.currentWorld
+    matches(action).when(NetworkWorldAction.userTyping.matches, ({ $from, typingDetail }) => {
+      const client = world.clients.get($from)
+      world.clients.set($from, JSON.parse(JSON.stringify({ ...client, typingDetail })))
+      setUserTypingDetail({
+        typing: typingDetail.typing,
+        userId: $from
+      })
+    })
+  }
```

Comment:
This would be simpler if we just defined a new state:

```ts
const UsersTypingState = defineState('usersTyping', () => [] as UserId[])

// in a system initializer
registerState(Engine.currentWorld.store,  UsersTypingState)

// in the AvatarDetailState component
const usersTyping = useState(getState(Engine.currentWorld.store, UsersTypingState))
```